test/parallel/test-sqlite-timeout.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+require('../common');
+const tmpdir = require('../common/tmpdir');
+const { path: fixturesPath } = require('../common/fixtures');
+const { spawn } = require('node:child_process');
+const { join } = require('node:path');
+const { DatabaseSync } = require('node:sqlite');
+const { test } = require('node:test');
+let cnt = 0;
+
+tmpdir.refresh();
+
+function nextDb() {
+  return join(tmpdir.path, `database-${cnt++}.db`);
+}
+
+test('connection can perform queries when lock is released before the timeout', (t, done) => {
+  const databasePath = nextDb();
+  const conn = new DatabaseSync(databasePath, { timeout: 1100 });
+  t.after(() => { conn.close(); });
+
+  conn.exec('CREATE TABLE data (key INTEGER PRIMARY KEY, value TEXT)');
+
+  // Spawns a child process to locks the database for 1 second
+  const child = spawn('./node', [fixturesPath('sqlite/lock-db.js'), databasePath], {
```

Comment:
Does this need to happen in a child process? Would it work to create a second database connection in the same process?

Reply:
I couldn't manage to get this done 😔 . We need:

1. to lock the `connection1`;
2. to try to perform a query using `connection2` with a busy timeout set;
3. to release lock hold in connection 1 before the timeout set for `connection2`.

In my attempts, I reached a scenario where either the `connection2` query waits for the whole timeout before the code that releases the lock can execute or the lock is released before the `connection2` query even starts.

That said, I agree it would be a lot better to have this in a single process. But I don't know how to do so. I'd appreciate any suggestions.